File: 57-insert-interval/insert-interval.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        ArrayList<int[]> list=new ArrayList<>();
        boolean added=false;
        if (intervals.length==0)
            return new int[][]{{newInterval[0],newInterval[1]}};
        for(int[] interval:intervals){
            if(interval[0]<newInterval[0] && !added){
                list.add(interval);
            }else{
                list.add(newInterval);                
                list.add(interval);

                added=true;
            }
        }
        if(!added){
            list.add(newInterval);
        }
        ArrayList<int[]> result=new ArrayList<>();
        result.add(list.get(0));
        for(int i=1;i<list.size();i++){
            int[] current=list.get(i);
            int[] prev=result.getLast();
            if(current[0]<=prev[1])
            {
                result.removeLast();
                result.add(new int[]{prev[0],Math.max(current[1],prev[1])});
            }else{
                result.add(current);
            }
        }
        int[][] r=new int[result.size()][2];
        result.toArray(r);
        return r;
    }
}

File: 57-insert-interval/insert-interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        inserted=False
        for i,interval in enumerate(intervals):
            start=interval[0]
            if(newInterval[0]<start):
                inserted=True
                intervals.insert(i, newInterval)
                break
        if not inserted:
            intervals.append(newInterval)
        result=[]
        index=0
        while index<len(intervals):
            start=intervals[index][0]
            end=intervals[index][1]
            while index<len(intervals) and intervals[index][0]<=end:
                end=max(end,intervals[index][1])
                index+=1
            result.append([start,end])

        return result